src/callbacks/update.jl

```diff
@@ -0,0 +1,133 @@
+struct UpdateCallback{I}
+    interval::I
+end
+
+"""
+    UpdateCallback(; interval::Integer, dt=0.0)
+
+Callback to update quantities either at the end of every `interval` integration step or at
+regular intervals at `dt` in terms of integration time.
+
+# Keywords
+- `interval`: Update quantities at the end of every `interval` time steps (default `interval=1`)
+- `dt`: Update quantities in regular intervals of `dt` in terms of integration time
```

Comment:
```suggestion
- `interval=1`: Update quantities at the end of every `interval` time steps.
- `dt`: Update quantities in regular intervals of `dt` in terms of integration time by adding additional `tstops` (note that this may change the solution).
```